src/main/java/com/example/demo/services/impl/UserServiceImpl.java

```diff
@@ -1,49 +1,70 @@
 package com.example.demo.services.impl;
 
-import com.example.demo.model.User;
+import com.example.demo.mappers.UserEntityMapper;
+import com.example.demo.mappers.UserMapper;
+import com.example.demo.model.dto.User;
+import com.example.demo.model.dto.UserCreation;
+import com.example.demo.model.dto.UserUpdate;
+import com.example.demo.model.entities.UserEntity;
+import com.example.demo.repositories.UserRepository;
 import com.example.demo.services.UserService;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
-import java.util.HashMap;
+import java.util.Date;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
 
 @Service
 public class UserServiceImpl implements UserService {
 
-    private final Map<String, User> users = new HashMap<>();
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    private final UserEntityMapper userEntityMapper;
+    public UserServiceImpl(UserRepository userRepository, UserMapper userMapper, UserEntityMapper userEntityMapper) {
+        this.userRepository = userRepository;
+        this.userMapper = userMapper;
+        this.userEntityMapper = userEntityMapper;
+    }
+
 
     @Override
-    public Map<String, User> getUsers() {
+    public List<User> getUsers() {
+        List<UserEntity> userEntities = userRepository.findAll();
+        List<User> users = userMapper.map(userEntities);
         return users;
     }
 
     @Override
-    public List<User> getUsersByAge(final int min, final int max) {
-        return users.values().stream()
-                .filter(user -> user.getAge() >= min && user.getAge() <= max)
-                .toList();
+    public List<User> getUsersByAgeRange(final int min, final int max) {
+        List<User> users = userRepository.findByAgeBetween(min, max);
+        return users;
     }
 
     @Override
-    public User getUserByName(final String name) {
-        return users.get(name);
+    public User getUserById(final long id) {
+        return userMapper.map(userRepository.getById(id));
     }
 
     @Override
-    public User addUser(final User user) {
-        users.put(user.getName(), user);
-        return user;
+    public String addUser(UserCreation user) {
+        userRepository.save(userEntityMapper.map(user));
+        return "User has been added";
```

Comment:
services are not responsible for the end message. Controllers should handle it.
You can change the signature of the methods for adding or updating users to `void` or `boolean` or `User`  that was updated or added. Probably, `User` is the most relevant for  `addUser` and `updateUser` and `boolean` or `void` for `deleteUser`